From e0d3a9e87961732b8d0d2616293c0d3287ae3c96 Mon Sep 17 00:00:00 2001
From: Zhou Jingyu <b02241@freescale.com>
Date: Wed, 4 May 2011 14:40:12 +0800
Subject: [PATCH 977/998] ENGR00142970 MX53: remove 160M working point

MX53 have no 160M working point any more. Remove 160M will
make mx53 spend more time in WFI mode. This patch can reduce
lpapm mode power to: VDDGP=10mw; VCC=35mw.

Signed-off-by: Zhou Jingyu <Jingyu.Zhou@freescale.com>
(cherry picked from commit fb2ed5be1709bf8d959e1888ee785b343b4c52ad)
---
 arch/arm/mach-mx5/bus_freq.c |    4 ++--
 arch/arm/mach-mx5/mx53_wp.c  |   28 ++++------------------------
 2 files changed, 6 insertions(+), 26 deletions(-)

diff --git a/arch/arm/mach-mx5/bus_freq.c b/arch/arm/mach-mx5/bus_freq.c
index bfac8dc..a4cd0d9 100644
--- a/arch/arm/mach-mx5/bus_freq.c
+++ b/arch/arm/mach-mx5/bus_freq.c
@@ -314,7 +314,7 @@ void enter_lpapm_mode_mx53()
 	/* if (mx53_ddr_type == DDR_TYPE_DDR2) {
 	} */
 
-	/* move cpu clk to pll2, 400 / 3 = 133Mhz for cpu  */
+	/* move cpu clk to pll2, 400 / 1 = 400MHZ for cpu  */
 	/* Change the source of pll1_sw_clk to be the step_clk */
 	reg = __raw_readl(MXC_CCM_CCSR);
 	reg |= MXC_CCM_CCSR_PLL1_SW_CLK_SEL;
@@ -324,7 +324,7 @@ void enter_lpapm_mode_mx53()
 	reg = __raw_readl(MXC_CCM_CDHIPR);
 	while (1) {
 		if ((reg & MXC_CCM_CDHIPR_ARM_PODF_BUSY) == 0) {
-			__raw_writel(0x2, MXC_CCM_CACRR);
+			__raw_writel(0x0, MXC_CCM_CACRR);
 			break;
 		} else {
 			reg = __raw_readl(MXC_CCM_CDHIPR);
diff --git a/arch/arm/mach-mx5/mx53_wp.c b/arch/arm/mach-mx5/mx53_wp.c
index cd3939e..a63bae4 100644
--- a/arch/arm/mach-mx5/mx53_wp.c
+++ b/arch/arm/mach-mx5/mx53_wp.c
@@ -50,18 +50,16 @@ static struct dvfs_wp dvfs_core_setpoint_ces_1_2G[] = {
 			{33, 25, 33, 10, 10, 0x08}, /*1_2GHz*/
 			{30, 18, 33, 20, 10, 0x08}, /* 800MHz */
 			{25, 8, 33, 20, 10, 0x08}, /* 400MHz */
-			{23, 0, 33, 20, 10, 0x08}, /* 160MHz */
-			{28, 8, 33, 20, 30, 0x08}, /*160MHz, 133MHz */
-			{29, 0, 33, 20, 10, 0x08},}; /* 160MHz, 50MHz. */
+			{28, 8, 33, 20, 30, 0x08}, /* 400MHZ, 133MHz */
+			{29, 0, 33, 20, 10, 0x08},}; /* 400MHZ, 50MHz. */
 
 /* Place holder for dvfs_core setpoints for 1 GHz parts */
 static struct dvfs_wp dvfs_core_setpoint_ces[] = {
 			{33, 25, 33, 10, 10, 0x08}, /*1GHz*/
 			{30, 18, 33, 20, 10, 0x08}, /* 800MHz */
 			{25, 8, 33, 20, 10, 0x08}, /* 400MHz */
-			{23, 0, 33, 20, 10, 0x08}, /* 160MHz */
-			{28, 8, 33, 20, 30, 0x08}, /*160MHz, 133MHz */
-			{29, 0, 33, 20, 10, 0x08},}; /* 160MHz, 50MHz. */
+			{28, 8, 33, 20, 30, 0x08}, /* 400MHz, 133MHz */
+			{29, 0, 33, 20, 10, 0x08},}; /* 400MHz, 50MHz. */
 
 /* working point for auto*/
 static struct cpu_wp cpu_wp_aec[] = {
@@ -105,15 +103,6 @@ static struct cpu_wp cpu_wp_ces[] = {
 	 .mfn = 1,
 	 .cpu_podf = 1,
 	 .cpu_voltage = 950000,},
-	{
-	 .pll_rate = 800000000,
-	 .cpu_rate = 160000000,
-	 .pdf = 0,
-	 .mfi = 8,
-	 .mfd = 2,
-	 .mfn = 1,
-	 .cpu_podf = 4,
-	 .cpu_voltage = 950000,},
 };
 
 /* working point for consumer 1.2G*/
@@ -150,15 +139,6 @@ static struct cpu_wp cpu_wp_ces_1_2g[] = {
 	  .cpu_rate = 400000000,
 	  .cpu_podf = 1,
 	  .cpu_voltage = 950000,},
-	{
-	 .pll_rate = 800000000,
-	 .cpu_rate = 160000000,
-	 .pdf = 0,
-	 .mfi = 8,
-	 .mfd = 2,
-	 .mfn = 1,
-	 .cpu_podf = 4,
-	 .cpu_voltage = 950000,},
 };
 
 static struct dvfs_wp *mx53_get_dvfs_core_table(int *wp)
-- 
1.7.4.1

